src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestPartials.qs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Testing.QIR
+{
+    open Microsoft.Quantum.Intrinsic;
+
+    @EntryPoint()
+    operation TestPartials () : Bool
+    {
+        let rotate = Rz(0.25, _);
+        let unrotate = Adjoint rotate;
+
+        for (i in 0..100)
+        {
+            using (qb = Qubit())
+            {
+                rotate(qb);
+                unrotate(qb);
```

Comment:
It would also be good to check that partial application after functor application works as expected, in particular, I'd check that both Controlled Op(_, (a, (b, c))) and Controlled Op(qs, (a, (_, c))) work as expected.